core/src/main/java/org/apache/iceberg/avro/PruneColumns.java

```diff
@@ -138,28 +166,98 @@ public Schema array(Schema array, Schema element) {
 
   @Override
   public Schema map(Schema map, Schema value) {
-    int keyId = AvroSchemaUtil.getKeyId(map);
-    int valueId = AvroSchemaUtil.getValueId(map);
+    Integer keyId = id(map, AvroSchemaUtil.KEY_ID_PROP, "key");
+    Integer valueId = id(map, AvroSchemaUtil.VALUE_ID_PROP, "value");
```

Comment:
I think the approach of adding a NameMapping lookup to the original `AvroSchemaUtil` methods would be a bit cleaner. What do you think?

Reply:
I ended up taking your suggestion.